nbformat/validator.py

```diff
@@ -291,17 +258,11 @@ def validate(nbdict=None, ref=None, version=None, version_minor=None,
         if version is None:
             version, version_minor = 1, 0
 
-    validator = get_validator(version, version_minor, relax_add_props=relax_add_props,
-                              use_fast=True)
-
-    if fastjsonschema is not None and use_fast:
-        if validator is None:
-            raise ValidationError("No schema for validating v%s notebooks" % version)
-
-        try:
-            validator(nbdict)
-        except JsonSchemaException as e:
-            raise ValidationError(e.message, schema_path=e.path)
+    validator = get_validator(version, version_minor, relax_add_props=relax_add_props)
+    if validator is None:
+        raise ValidationError("No schema for validating v%s notebooks" % version)
+    elif use_fast:
```

Comment:
no longer needed?

Reply:
No, actually it is needed, otherwise the logic will break :-|

Reply:
Since the better_validation_errors needs to use the iter_iterate so there needs to be a switch.

Reply:
I will think a bit harder on decoupling this logic 🤔